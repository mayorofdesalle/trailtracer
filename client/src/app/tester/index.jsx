import '@testing-library/jest-dom';

import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';

import './mocks/i18n.mock';
import './mocks/matchMedia.mock';
import './mocks/resizeObserver.mock';

import store from '@app/store';
import GlobalStyle from '@components/style/GlobalStyle';
import Theme from '@components/style/Theme';

const Providers = ({ children }) => {
    return (
        <Provider store={store}>
            <Theme>
                <GlobalStyle />
                <BrowserRouter>
                    {children}
                </BrowserRouter>
            </Theme>
        </Provider>
    );
};

const customRender = (ui, options) =>
    render(ui, { wrapper: Providers, ...options });

// re-export everything
export * from '@testing-library/react';
export * from 'vitest';

// override render method
export { customRender as render };
