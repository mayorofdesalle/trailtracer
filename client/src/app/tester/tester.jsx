import { render } from '@testing-library/react'
import { Provider } from 'react-redux';

import './mocks/matchMedia.mock';
import store from '@app/store';
import Theme from '@components/style/Theme'

const Providers = ({ children }) => {
    return (
        <Provider store={store}>
            <Theme>
                {children}
            </Theme>
        </Provider>
    );
};

const customRender = (ui, options) =>
    render(ui, { wrapper: Providers, ...options });

// re-export everything
export * from '@testing-library/react';
export * from 'vitest';

// override render method
export { customRender as render };