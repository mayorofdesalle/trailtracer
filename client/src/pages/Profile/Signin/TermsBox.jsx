import { useContext } from 'react';
import styled from 'styled-components';

import BentoBox from '@components/ui/Bento/BentoBox';
import Icon from '@components/ui/Icon';
import Text from '@components/ui/Text';
import PageContext from '@context/PageContext';

const TermsBoxInner = styled(BentoBox)`
    grid-area: TermsBox;
    padding: 8%;
    flex-direction: column;
    justify-content: space-around;

    & > svg {
        height: 4dvh;
        width: 4dvh;
    }
`;

const TermsBox = () => {
    const context = useContext(PageContext);
    const theme = context.theme;
    const t = context.t;
    const largeScreen = context.largeScreen;

    return (
        <TermsBoxInner $glass>
            {largeScreen && <Icon name='article-fill' color={theme.colors.primary} />}
            <Text $color={theme.colors.text} $ratio={0.9}>
                {t('signinPage.termsPrefix')} <a href=''>{t('signinPage.terms')}</a> {t('signinPage.termsSuffix')}
            </Text>
        </TermsBoxInner>
    );
};

export default TermsBox;