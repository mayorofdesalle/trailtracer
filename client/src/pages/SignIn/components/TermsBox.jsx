import styled from 'styled-components';

import BentoBox from '@components/ui/Bento/Box';

const TermsBox = styled(BentoBox)`
    grid-area: TermsBox;
`;

export default TermsBox;