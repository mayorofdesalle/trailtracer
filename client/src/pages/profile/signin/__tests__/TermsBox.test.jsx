import { describe, expect, it, render, screen } from '@app/tester';
import userEvent from '@testing-library/user-event';

import TermsBox from '../TermsBox';

describe('TermsBox', () => {

    it('should render correctly', () => {
        render(<TermsBox />);
        const termsBox = screen.getByRole('note');

        expect(termsBox).toBeInTheDocument();
        expect(termsBox).toHaveStyle({ gridArea: 'TermsBox' });
        expect(termsBox).toHaveTextContent(/terms/i);

        const icon = screen.getByRole('img');
        expect(icon).toBeInTheDocument();

        const link = screen.getByRole('link');
        expect(link).toBeInTheDocument();
    });

    it('should match snapshot', () => {
        const { asFragment } = render(<TermsBox />);
        expect(asFragment()).toMatchSnapshot();
    });

});