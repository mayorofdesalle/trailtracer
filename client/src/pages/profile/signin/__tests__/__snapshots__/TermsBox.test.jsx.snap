// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TermsBox > should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-guDLey sc-dmyCSP sc-irLvIq kHwwsn dHdhmG lfDmDW"
    role="note"
  >
    <svg
      class="sc-eDLKkx fuXsHX"
      fill="#435049"
      role="img"
    >
      <title>
        article-fill
      </title>
      <use
        href="http://localhost:3000/src/assets/icons.svg#article-fill"
      />
    </svg>
    <div
      class="sc-gLLuof csXMhK"
    >
      signinPage.termsPrefix 
      <a
        href=""
      >
        signinPage.terms
      </a>
       signinPage.termsSuffix
    </div>
  </div>
</DocumentFragment>
`;
